chromium_src/chrome/test/views/chrome_views_test_base.cc

```diff
@@ -0,0 +1,41 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "chrome/test/views/chrome_views_test_base.h"
+
+#include "ui/gfx/animation/animation_test_api.h"
+
+#define ChromeViewsTestBase ChromeViewsTestBase_ChromiumImpl
+
+#include "src/chrome/test/views/chrome_views_test_base.cc"
+
+#undef ChromeViewsTestBase
+
+ChromeViewsTestBase::ChromeViewsTestBase() = default;
+
+ChromeViewsTestBase::~ChromeViewsTestBase() = default;
+
+void ChromeViewsTestBase::SetUp() {
+  if (const testing::TestInfo* test_info =
+          testing::UnitTest::GetInstance()->current_test_info()) {
+    // Force animations to be always enabled to not fail some upstream tests
+    // when run under RDP session.
+    static constexpr base::StringPiece kRichAnimationForcedSuites[] = {
+        "CompoundTabContainerTest",
+        "TabContainerTest",
+    };
+    if (base::Contains(kRichAnimationForcedSuites,
+                       test_info->test_suite_name())) {
+      animation_mode_reset_ = gfx::AnimationTestApi::SetRichAnimationRenderMode(
+          gfx::Animation::RichAnimationRenderMode::FORCE_ENABLED);
```

Comment:
Some of upstream tests are setting FORCE_DISABLED, would they be okay? e.g. https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/sessions/tab_restore_browsertest.cc;l=73;drc=1187260c8a04c6608f6605d6d25678399c7e467f;bpv=1;bpt=1

Maybe we can make ChromeViewsTestBaseWithRichAnimation and replace onlyl the base class of CompoundTabContainerTest and TabContainerTest.

Reply:
> Some of upstream tests are setting FORCE_DISABLED, would they be okay? e.g. https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/sessions/tab_restore_browsertest.cc;l=73;drc=1187260c8a04c6608f6605d6d25678399c7e467f;bpv=1;bpt=1
> 
> Maybe we can make ChromeViewsTestBaseWithRichAnimation and replace onlyl the base class of CompoundTabContainerTest and TabContainerTest.

this only overrides `CompoundTabContainerTest` and `TabContainerTest` tests using `test_suite_name`.

I tried to force this globally, but as you noted, other tests may break (and they break, yep).

Reply:
Aha! That's it! Brilliant. Thanks for the explanation 👍